cs/src/core/Index/FASTER/FASTERLegacy.cs

```diff
@@ -160,7 +160,8 @@ public Status Read(ref Key key, ref Input input, ref Output output, Context cont
         [Obsolete("Use NewSession() and invoke Upsert() on the session.")]
         public Status Upsert(ref Key key, ref Value value, Context context, long serialNo)
         {
-            return _fasterKV.ContextUpsert(ref key, ref value, context, FasterSession, serialNo, _threadCtx.Value);
+            PSFUpdateArgs<Key, Value> psfUpdateArgs = default;
```

Comment:
Remove - legacy does not need to support PSF. Also ContextUpsert etc. will not have extra argument.